src/runtime/services.ts

```diff
@@ -7,12 +7,32 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+// TODO(alxr): should this go into `functional` or a "types" file?
+interface Dictionary<T> {
+  [key: string]: T;
+}
+
+interface Service {
+  [name: string]: Function;
```

Comment:
Can we narrow this type? e.g. Annotate the function's type?

Reply:
Good question. I had a discussion with @sjmiles about this the other day. 

So far, we want to have as wide-open of "constraints" as possible while we are prototyping. Another way of saying this: Until we figure out what the service functions minimally need to consume/return, we're going to type this as generically as possible.

Reply:
I had been thinking about something like:

```
interface Service <A extends Function> {
     [name: string]: A;
}
```
This would allow us to have a `Service<(a: number, b:number) => number[]>`, but the problem with this is that it will not allow multiple different types for functions inside a service.
Ideally we'd have more specific types that know about the function names (and their mapping to types) inside the Service, but that's probably not needed here either.

So, in summary if we need the flexibility of multiple types inside the service then go ahead with your approach, but I will warn of the potential for errors that this will create (that the type system could have caught if we gave it more information).

Reply:
That's an interesting idea. If developers wanted to go this way, they could create unions of function types -- though, this could become cumbersome. 

Another option is developers can subclass the `Service` interface and type each individual function.